workflows/src/test/kotlin/com/r3/corda/lib/tokens/workflows/ConfigSelectionTest.kt

```diff
@@ -0,0 +1,183 @@
+            ConfigSelection.getPreferredSelection(services, cordappConfigBlank)
+        }
+    }
+
+    @Test
+    fun `test defaults database selection`() {
+        val configOne = ConfigFactory.parseString("stateSelection {\n" +
+                "database {\n" +
+                "maxRetries: 13\n" +
+                "retryCap: 1345\n" +
+                "}\n" +
+                "}")
+        val cordappConfigOne = TypesafeCordappConfig(configOne)
+        val selectionOne = ConfigSelection.getPreferredSelection(services, cordappConfigOne)
+        assertThat(selectionOne).isInstanceOf(TokenSelection::class.java)
+        assertThat(selectionOne).hasFieldOrPropertyWithValue("maxRetries", 13)
+        assertThat(selectionOne).hasFieldOrPropertyWithValue("retrySleep", RETRY_SLEEP_DEFAULT)
+        assertThat(selectionOne).hasFieldOrPropertyWithValue("retryCap", 1345)
+
+        val configAll = ConfigFactory.parseString("stateSelection {\n" +
+                "database {}\n" +
+                "}")
+        val cordappConfigAll = TypesafeCordappConfig(configAll)
+        val selectionAll = ConfigSelection.getPreferredSelection(services, cordappConfigAll)
+        assertThat(selectionAll).isInstanceOf(TokenSelection::class.java)
+        assertThat(selectionAll).hasFieldOrPropertyWithValue("maxRetries", MAX_RETRIES_DEFAULT)
+        assertThat(selectionAll).hasFieldOrPropertyWithValue("retrySleep", RETRY_SLEEP_DEFAULT)
+        assertThat(selectionAll).hasFieldOrPropertyWithValue("retryCap", RETRY_CAP_DEFAULT)
+    }
+
+    @Test
+    fun `test defaults in memory selection`() {
+        createMockCordaService(services, ::VaultWatcherService)
+        val config = ConfigFactory.parseString("stateSelection {\n" +
+                "inMemory {\n" +
+                "}\n" +
+                "}")
+        val cordappConfig = TypesafeCordappConfig(config)
+        val inMemoryConfig = InMemorySelectionConfig.parse(cordappConfig)
+        assertThat(inMemoryConfig.cacheSize).isEqualTo(CACHE_SIZE_DEFAULT)
+        assertThat(inMemoryConfig.indexingStrategy).isEqualTo(VaultWatcherService.IndexingType.PUBLIC_KEY)
+    }
+
+    @Test
+    fun `test fail database selection`() {
+        val config = ConfigFactory.parseString("stateSelection {\n" +
+                "database {\n" +
+                "maxRetries: 13\n" +
+                "retrySleep: kerfuffle\n" +
+                "retryCap: 1345\n" +
+                "}\n" +
+                "}")
+        val cordappConfig = TypesafeCordappConfig(config)
+        assertThatExceptionOfType(IllegalArgumentException::class.java).isThrownBy {
+            ConfigSelection.getPreferredSelection(services, cordappConfig)
+        }
+    }
+
+    @Test
+    fun `test fail in memory selection`() {
+        val config = ConfigFactory.parseString("stateSelection {\n" +
+                "inMemory {\n" +
+                "cacheSize: kerfuffle\n" +
```

Comment:
uh oh!

Reply:
this is to check that you actually read the whole code :joy: